api/v1alpha3/virtualmachine_types.go

```diff
@@ -449,6 +449,12 @@ type VirtualMachineSpec struct {
 	// +optional
 	// +kubebuilder:validation:Format:=uuid
 
+	// InstanceUUID describes the desired Instance UUID for a VM.
+	// If omitted, this field defaults to a random UUID.
+	InstanceUUID string `json:"instanceUUID,omitempty"`
```

Comment:
I would also add an additional comment indicating this is *not* the same as any potential Cloud-Init instance ID, briefly describing the difference between the two fields.

Reply:
Done